runbotics/runbotics-desktop/src/action/desktop/desktop.action-handler.ts

```diff
@@ -0,0 +1,124 @@
+import { StatelessActionHandler } from 'runbotics-sdk';
+import { RunboticsLogger } from '#logger';
+import {
+    mouse,
+    keyboard,
+    straightTo,
+    Point,
+    Button,
+    Key,
+} from '@nut-tree/nut-js';
+import * as DesktopTypes from './types';
+
+export default class DesktopActionHandler extends StatelessActionHandler {
+    private readonly logger = new RunboticsLogger(DesktopActionHandler.name);
+
+    constructor() {
+        super();
+    }
+
+    private checkIfNaN(xCoordinate: number, yCoordinate: number) {
+        if (isNaN(xCoordinate) || isNaN(yCoordinate)) {
+            throw new Error('Element coordinates have to be of type number');
+        }
+    }
+
+    async click(input: DesktopTypes.DesktopClickActionInput) {
+        //this.checkIfNaN(input.xCoordinate, input.yCoordinate);
+        mouse.config = {
+            autoDelayMs: 20,
+            mouseSpeed: 500,
+        };
+
+        await mouse.move(
+            straightTo(new Point(input.xCoordinate, input.yCoordinate))
+        );
+        if (input.mouseButton === DesktopTypes.MouseButton.RIGHT) {
+            await mouse.click(Button.RIGHT);
+        } else {
+            await mouse.click(Button.LEFT);
+        }
+    }
+
+    async type(input: DesktopTypes.DesktopTypeActionInput) {
+        // if not of type string then make string
+        // this.checkIfNaN(input.xCoordinate, input.yCoordinate);
+        mouse.config = {
+            autoDelayMs: 50,
+            mouseSpeed: 500,
+        };
+        keyboard.config = {
+            autoDelayMs: 50,
+        };
+
+        await mouse.move(
+            straightTo(new Point(input.xCoordinate, input.yCoordinate))
+        );
```

Comment:
you're using this >2 times

consider creating method
async moveMouse(x: string, y:string): void {
straightTo(new Point(x, y))
}